lib/vssh/libssh.c

```diff
@@ -1453,7 +1453,9 @@ static CURLcode myssh_statemach_act(struct Curl_easy *data, bool *block)
           if((sshc->readdir_attrs->flags & SSH_FILEXFER_ATTR_PERMISSIONS) &&
              ((sshc->readdir_attrs->permissions & S_IFMT) ==
               S_IFLNK)) {
-            sshc->readdir_linkPath = malloc(PATH_MAX + 1);
+            sshc->readdir_linkPath = malloc(strlen(protop->path) +
```

Comment:
several lines below this it uses PATH_MAX. (i can't comment on the individual line because it's a common line not add/remove)

Reply:
Oh right! I forgot, let me correct it

Reply:
Fixed it!